### ISSUE TITLE
MigrateClients mode throws exception (System.InvalidOperationException: Reading is already in progress.System.InvalidOperationException: Reading is already in progress.)

### ISSUE THREAD

**Issue by chertpong** [Azure/azure-signalr#1218]
### Describe the bug
I tested out how `GracefulShutdownMode.MigrateClients` works and hit the exception at the new server where connection is migrated to.

### To Reproduce
Here is the steps
1) Start 2 servers behind the load balancer.
2) Use javascript client to connect to one of them
3) Client is connected to server1
4) Shutdown server1
5) The connection didn't reach `OnConnectedAsync()` in Hub and throws exception
6) Client disconnected and rejoin as new connection. // it supposed to be migrated not disconnected

https://github.com/chertpong/sample-signalr

### Exceptions (if any)
```log
fail: Microsoft.Azure.SignalR.ServiceConnection[23]
      Error while skipping handshake response during migration, the connection will be dropped on the client-side. Error detail: Reading is already in progress.
System.InvalidOperationException: Reading is already in progress.
   at System.IO.Pipelines.ThrowHelper.ThrowInvalidOperationException_AlreadyReading()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.SignalR.ServiceConnection.SkipHandshakeResponse(ClientConnectionContext connection, CancellationToken token)
```

### Further technical details
- Your Azure SignalR SDK version
1.6.1
- Your Server ASPNETCORE version or Assembly version of `Microsoft.AspNetCore.SignalR` 
3.1.401
- Your SignalR Client SDK version
I'm not sure about the version since I'm using the signalr.js from sample project

**Comment by terencefan** [comment]
Acknowledged, problem reproduced, and currently working on this.

**Comment by terencefan** [comment]
This feature should have been fixed in `1.10.0`
#1376
